Ginger/Ginger/UserControlsLib/UCDataMapping.xaml.cs

```diff
@@ -0,0 +1,553 @@
+            SetValueControlsData();
+        }
+        #endregion Variables
+
+        #region Output Variables
+        private static void OnOutputVariabelsSourcePropertyChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+        {
+            var control = sender as UCDataMapping;
+            if (control != null)
+            {
+                control.OutputVariabelsSourcePropertyChanged((ObservableList<VariableBase>)args.NewValue);
+            }
+        }
+        private void OutputVariabelsSourcePropertyChanged(ObservableList<VariableBase> outputVariabelsSourceList)
+        {
+            if (outputVariabelsSourceList == null || outputVariabelsSourceList.Count == 0)
+            {
+                GingerCore.General.DisableComboItem(xMappedTypeComboBox, eDataType.OutputVariable);
+            }
+           
+            if (outputVariabelsSourceList != null && outputVariabelsSourceList.Count > 0)
+            {
+                if (EnableDataMapping)
+                {
+                    GingerCore.General.EnableComboItem(xMappedTypeComboBox, eDataType.OutputVariable);
+                }
+                mOutputVariablesList = outputVariabelsSourceList;
+                mOutputVariablesList.CollectionChanged += OutputVariabelsSourceList_CollectionChanged;
+                SetValueControlsData();
+            }            
+        }
+        private void OutputVariabelsSourceList_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+        {
+            SetValueControlsData();
+        }
+        #endregion Output Variables
+
+        #region Global Variabels
+        private void SetGlobalVariabelsListValues()
+        {
+            mGlobalVariablesList = new ObservableList<VariableBase>();
+            WorkSpace.Instance.Solution.Variables.CollectionChanged += GlobalVariables_CollectionChanged;
+            foreach (VariableBase var in WorkSpace.Instance.Solution.Variables.Where(x => x.SupportSetValue == true).ToList())
+            {
+                mGlobalVariablesList.Add(var);
+            }
+
+            if (mGlobalVariablesList.Count > 0)
+            {
+                GingerCore.General.EnableComboItem(xMappedTypeComboBox, eDataType.GlobalVariable);
+            }
+        }
+
+        private void GlobalVariables_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+        {
+            //update list
+            foreach (VariableString var in WorkSpace.Instance.Solution.Variables.Where(x => x is VariableString).ToList())
+            {
+                if (mGlobalVariablesList.Contains(var) == false)
+                {
+                    mGlobalVariablesList.Add(var);
+                }
+            }
+            for (int indx=0; indx < mGlobalVariablesList.Count; indx++)
+            {
+                if (WorkSpace.Instance.Solution.Variables.Contains(mGlobalVariablesList[indx]) == false)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5538205)